### ISSUE TITLE
[Xcode 8.3.3] Could not install WWDR certificate

### ISSUE THREAD

**Issue by romk1n** [travis-ci/travis-ci#7870]
All builds are failing today since the 8.3 image has been updated to 8.3.3 and now when executing `fastlane match` we getting this error:

```
[15:00:36]: Could not install WWDR certificate
```

What's the workaround?

**Comment by BanzaiMan** [comment]
Do you have a build log URL that shows the problem you are describing here?

**Comment by romk1n** [comment]
@BanzaiMan it's a private repo so i don't think i can paste the full log but all builds worked yesterday with Xcode 8.3.2

```
+-----------------------------+---------------------------+
|                      Lane Context                       |
+-----------------------------+---------------------------+
| DEFAULT_PLATFORM            | ios                       |
| PLATFORM_NAME               | ios                       |
| LANE_NAME                   | ios build_development_ipa |
| GIT_REPO_WAS_CLEAN_ON_START | true                      |
+-----------------------------+---------------------------+
[15:00:36]: Could not install WWDR certificate
+------+-------------------------------------+-------------+
|                     fastlane summary                     |
+------+-------------------------------------+-------------+
| Step | Action                              | Time (in s) |
+------+-------------------------------------+-------------+
| 1    | Verifying required fastlane version | 0           |
| 2    | is_ci                               | 0           |
| 3    | is_ci                               | 0           |
| 4    | is_ci                               | 0           |
| 5    | default_platform                    | 0           |
| 6    | Switch to ios prepare lane          | 0           |
| 7    | ensure_xcode_version                | 1           |
| 8    | ensure_git_status_clean             | 0           |
| 9    | Switch to ios match_cmd lane        | 0           |
| 💥    | match                               | 13          |
+------+-------------------------------------+-------------+
[15:00:36]: fastlane finished with errors
[!] Could not install WWDR certificate
```

**Comment by BanzaiMan** [comment]
Is fastlane trying to install WWDR certificate?

**Comment by BanzaiMan** [comment]
I can read the build log if you provide the URL. If you do not wish to discuss this in public, please send email to support@travis-ci.com with details, please.

**Comment by romk1n** [comment]
@BanzaiMan sent email thanks!

**Comment by BanzaiMan** [comment]
Here's what I found:

The error manifests because of the way fastlane manages the WWDRCA certificate.

See https://github.com/fastlane/fastlane/blob/1240fe39eaf236eb9774ec985a437b123cb10aa7/fastlane_core/lib/fastlane_core/cert_checker.rb#L64-L65

It attempts to download https://developer.apple.com/certificationauthority/AppleWWDRCA.cer and install it in a keychain that the user has access to. It is assumed here that the nonzero exit status means the *failure* to import the certificate, but it does not account for the possibility the nonzero exit because the key already exists.

I believe the error manifests now on Travis CI, because we added the WWDRCA certificate to our login keychain.

As I indicated earlier, the real fix should come from the fastlane team, but in the mean time, I believe that removing the certificate fastlane expects to add itself would work.

```yaml
before_install:
  - security delete-certificate -c "Apple Worldwide Developer Relations Certification Authority"
  ⋮ # the rest using fastlane
```
